posthog/api/authentication.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any
+
+from django.conf import settings
+from rest_framework import generics, permissions, serializers
+
+from posthog.settings import print_warning
+
+
+class AuthenticationSerializer(serializers.Serializer):
+    available_backends = serializers.SerializerMethodField()
+
+    def get_available_backends(self, *args):
+        github: bool = bool(settings.SOCIAL_AUTH_GITHUB_KEY and settings.SOCIAL_AUTH_GITHUB_SECRET)
+        gitlab: bool = bool(settings.SOCIAL_AUTH_GITLAB_KEY and settings.SOCIAL_AUTH_GITLAB_SECRET)
+        google: bool = False
+
+        if getattr(settings, "SOCIAL_AUTH_GOOGLE_OAUTH2_KEY", None) and getattr(
+            settings, "SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET", None,
+        ):
+            if settings.MULTI_TENANCY:
+                google = True
+            else:
+
+                try:
+                    from ee.models.license import License
+                except ImportError:
+                    pass
+                else:
+                    license = License.objects.first_valid()
+                    if license is not None and "google_login" in license.available_features:
+                        google = True
+                    else:
+                        print_warning(["You have Google login set up, but not the required premium PostHog plan!"])
+
+        return {"google-oauth2": google, "github": github, "gitlab": gitlab}
```

Comment:
This is very purpose-specific. Makes sense here, but we already have quite a bit of endpoints returning something pertaining globally to the instance and I'm afraid we'll be left with a mess if every feature gets its own endpoint. Could we roll this into some more global (and still lightweight) instance capabilities endpoint?

Reply:
Yup thought about that, was a bit skeptical of using the preflight endpoint (which seems like the best match) because it wasn't on DRF, but after some thought, it should be fine, and it's also a fairly simple endpoint to migrate too.

Reply:
Semantically I'd like to use something like `/api/instance/capabilities`, but that's not really important and preflight should be a very appropriate place I think